src/backends/torch/torchinputconns.cc

```diff
@@ -942,22 +956,31 @@ namespace dd
     std::vector<std::vector<CSVline>> &data
         = test ? this->_csvtsdata_test : this->_csvtsdata;
 
-    for (const std::vector<CSVline> &seq : data)
+    if (_train)
       {
-        vecindex++;
-        long int tstart = 0;
-        if (static_cast<long int>(seq.size()) < _timesteps)
+        for (const std::vector<CSVline> &seq : data)
           {
-            discard_warn(vecindex, seq.size(), test);
-            continue;
+            vecindex++;
+            long int tstart = 0;
+            if (static_cast<long int>(seq.size()) < _timesteps)
+              {
+                discard_warn(vecindex, seq.size(), test);
+                continue;
+              }
+            for (; tstart + _timesteps < static_cast<long int>(seq.size());
+                 tstart += _offset)
+              add_data_instance_labels(tstart, vecindex, dataset, seq);
+            if (tstart < static_cast<long int>(seq.size()) - 1)
+              add_data_instance_labels(seq.size() - _timesteps, vecindex,
+                                       dataset, seq);
```

Comment:
So if sequence length is 500, timestep is 200 and offset is 200, the training sequences will be [0, 200], [200, 400] and [300, 500] ? The last sequence seems arbitrary, maybe this behavior should be referenced somewhere?

Reply:
this is the best way I could think of for taking into account all data (ie do not discard [400, 500] in your example). I added a sentence in api.md